ddtrace/contrib/langchain/patch.py

```diff
@@ -677,41 +717,83 @@ def patch():
 
     # Langchain doesn't allow wrapping directly from root, so we have to import the base classes first before wrapping.
     # ref: https://github.com/DataDog/dd-trace-py/issues/7123
-    from langchain import embeddings  # noqa:F401
-    from langchain import vectorstores  # noqa:F401
-    from langchain.chains.base import Chain  # noqa:F401
-    from langchain.chat_models.base import BaseChatModel  # noqa:F401
-    from langchain.llms.base import BaseLLM  # noqa:F401
-
-    wrap("langchain", "llms.base.BaseLLM.generate", traced_llm_generate(langchain))
-    wrap("langchain", "llms.base.BaseLLM.agenerate", traced_llm_agenerate(langchain))
-    wrap("langchain", "chat_models.base.BaseChatModel.generate", traced_chat_model_generate(langchain))
-    wrap("langchain", "chat_models.base.BaseChatModel.agenerate", traced_chat_model_agenerate(langchain))
-    wrap("langchain", "chains.base.Chain.__call__", traced_chain_call(langchain))
-    wrap("langchain", "chains.base.Chain.acall", traced_chain_acall(langchain))
+    if SHOULD_USE_LANGCHAIN_COMMUNITY:
+        from langchain.chains.base import Chain  # noqa:F401
+        from langchain_community import embeddings  # noqa:F401
+        from langchain_community import vectorstores  # noqa:F401
+        from langchain_core.language_models.chat_models import BaseChatModel  # noqa:F401
+        from langchain_core.language_models.llms import BaseLLM  # noqa:F401
```

Comment:
Do we need these imports? If they are not used in patch should we just move them to `unpatch`?

Reply:
The last two you're right I can just remove them, but the chain/embeddings/vectorstore classes need to be imported first. This is because langchain doesn't allow wrapping directly from root, so we have to import the base classes first before wrapping.